api/clients/mock/retrieval_client.go

```diff
@@ -35,3 +35,21 @@ func (c *MockRetrievalClient) RetrieveBlob(
 	result := args.Get(0)
 	return result.([]byte), args.Error(1)
 }
+
+func (c *MockRetrievalClient) RetrieveBlobChunks(
+	ctx context.Context,
+	batchHeaderHash [32]byte,
+	blobIndex uint32,
+	referenceBlockNumber uint,
+	batchRoot [32]byte,
+	quorumID core.QuorumID) (*clients.BlobChunks, error) {
+
```

Comment:
Do we want `args := c.Called()` and pass return values?
I guess this method isn't being used in any of test cases rn so wouldn't matter, but would be good to implement it instead of always returning nil

Reply:
I had to fix this in order to get my tests working (after deleting the duplicate implementation of the mock retrieval client).